### PR TITLE
Update to Python-3.8

### PR DESCRIPTION
**Pull request by qbarthelemy**
Update to Python-3.8

Update code to support SciPy 1.11.0 and scikit-learn 1.3.0.

Update readme for version 0.5, see #253.

### CODE REVIEW

**Comment by qbarthelemy** [comment]
`scipy.stats.mode` for non numeric data is deprecated from SciPy 1.9.0 and is removed in SciPy 1.11.0.

Tests are broken, because transfer learning module uses non numeric extended labels. 

@plcrodrigues , any idea?

**Comment by qbarthelemy** [comment]
I solve the problem, replacing `scipy.stats.mode` by `np.unique`, which supports non numeric data.

**Comment by plcrodrigues** [comment]
> @plcrodrigues , any idea?

This is indeed annoying but doesn't sound too hard to bypass with your `np.unique` approach.

In any case, we should soon take some time to make use of the new `sample-props` feature from `sklearn` -- [PR #24027](https://github.com/scikit-learn/scikit-learn/pull/24027) -- in the transfer learning module of `pyRiemann`. This would avoid recurring to string in the labels.

File: pyriemann/classification.py [hunk 1]
```diff
@@ -524,8 +523,8 @@ def predict(self, covtest):
         """
         dist = self._predict_distances(covtest)
         neighbors_classes = self.classmeans_[np.argsort(dist)]
-        out, _ = stats.mode(neighbors_classes[:, 0:self.n_neighbors], axis=1)
-        return out.ravel()
+        out = np.unique(neighbors_classes[:, 0:self.n_neighbors], axis=1)[:, 0]
```
Comment (agramfort) [inline pyriemann/classification.py:526]: I doubt this is correct

Comment (agramfort) [reply pyriemann/classification.py:526]: you would need to use return_counts and get the entry in out that has the biggest count

Comment (qbarthelemy) [reply pyriemann/classification.py:526]: You are right. Solution is a little bit more complicated
https://stackoverflow.com/a/12300214

I will try to find the simplest solution for 2D case.

**Comment by qbarthelemy** [comment]
Last test for Windows Python 3.7 which is OK installed joblib 1.2.0
https://github.com/pyRiemann/pyRiemann/actions/runs/5321719371/jobs/9637189077#step:5:24
[5:10](https://bciparis.slack.com/archives/G010MU5D3UH/p1688137804132769)

Now, it is KO installing joblib 1.3.1
https://github.com/pyRiemann/pyRiemann/actions/runs/5415226192/jobs/9846294273?pr=254#step:5:24

I freeze joblib to 1.2.0, but now, all others tests for Python 3.8 and 3.9 are KO.

Any idea?

**Comment by qbarthelemy** [comment]
Waiting scikit-learn bugfix, I decide to detect when _init_centroids must be used with a non default sample_weight.
I also clean code, removing support for version before scikit-learn 0.24.

**Comment by lesteve** [comment]
FYI I had a look and the joblib 1.3.1 issue is actually due to loky (default joblib parallel backend). The loky issue is tracked in https://github.com/joblib/loky/issues/411.

Not entirely sure this is going to get fixed in loky. FYI scikit-learn requires Python 3.8+ since scikit-learn 1.1 (released May 2022), so a possible work-around for pyRiemann would be to drop Python 3.7 support in the 0.5 release.

**Comment by qbarthelemy** [comment]
Close-reopen for tests.

**Comment by qbarthelemy** [comment]
Documentation fails, it is probably related to #255. @agramfort, what do you think ?

**Review by agramfort — APPROVED** [review]
thx @qbarthelemy

**Comment by qbarthelemy** [comment]
For documentation build, I can't solve the dependency problem between sphinx and jinja2.

**Comment by qbarthelemy** [comment]
Doc is now OK!
I let you merge @agramfort , @plcrodrigues , @sylvchev

**Review by agramfort — APPROVED** [review]
